### ISSUE TITLE
IMPORTANT: A propose to include the rough roadmap introduced by Charles Hoskinson from IOHK (decision timeframe: till end of the 14th of Aug'20)

### ISSUE THREAD

**Issue by TheEnthusiasticAs** [ethereumclassic/ECIPs#331]
**Include the rough roadmap introduced by Charles Hoskinson from IOHK for the ETC blockchain into the current roadmap - partially or fully.**

I am asking the ETC participants to watch this video by Charles Hoskinson from IOHK, produced yesterday, on 9th of August 2020, in regards of latest chain reorganization (3rd one). He proposes as well a short term solution as a long term for the ETC blockchain project. The video: https://youtu.be/oHUQuXOwYeU

**A short summary of the solutions:**
Short term solution:
_Securing Proof-of-Work Ledgers via Checkpointing:
https://iohk.io/en/research/library/papers/securing-proof-of-work-ledgers-via-checkpointing/_
Long term solution for the project development:
_A Proposal for An Ethereum Classic Treasury System:
https://iohk.io/en/research/library/papers/a-proposal-for-an-ethereum-classic-treasury-system/_

The decision should be made till 14th of August 2020. I took a reserve of 2 days. The deadline is 16th of August (1 week. It is mentioned in the video).

Instead of the checkpointing also e.g. this solution #328 (PirlGuard) can be taken ("copy-pasteable") as a short term solution to "protect" the network "against" reorganizations as with the current provided hashrates the network can be reorganized at any time. The network gets unusable for the network participants.

Please comment below. Thank you!

**Comment by crazypool2019** [comment]
pirlguard + checkpoints

**Comment by TheEnthusiasticAs** [comment]
ok, what about to include the roadmap, without the checkpointing, @crazypool2019 ?

**Comment by BaikalMine** [comment]
I suggest using PirlGuard. This is the simplest, fastest and I find it reliable. Moreover, it can be turned off at any time if after a while the same reliable solution is presented from the ETC developers themselves.

**Comment by Dexaran** [comment]
I'm committing my support towards the Treasury model.

However, I think that instead of developing the implementation at a protocol level the Treasury system can be implemented in smart-contract layer with minimal changes to the protocol. I had success with smart-contract layer solutions in past.

**Comment by Dexaran** [comment]
I would propose to implement PirlGuard first as it can effectively solve the problem right now. The network is barely operational and this needs to be fixed.

It should be noted that PirlGuard can be enabled and disabled with minimal efforts which makes it an ideal temporary solution until long-term proposals are carefully evaluated and time-consuming resrach/debates are conducted.

**Comment by TheEnthusiasticAs** [comment]
> I'm committing my support towards the Treasury model.
> 
> However, I think that instead of developing the implementation at a protocol level the Treasury system can be implemented in smart-contract layer with minimal changes to the protocol. I had success with smart-contract layer solutions in past.

@Dexaran In the treasury concept by IOHK there were mentioned, iirc, that the solution can also be done on the side chain or as a smart contract. The video: https://www.youtube.com/watch?v=zxZoSjnHN84

**Comment by BaikalMine** [comment]
I have not seen the code for this solution, so I believe that this solution is not suitable in the current situation. PirlGuard is our everything.

**Comment by TheEnthusiasticAs** [comment]
For the short term, I support that solution, which in fewest change the ETC blockchain's principles, if there are some.
The idea of the treasury system, I support, too - if possible as a side chain solution (layer 2). 
To include the rough roadmap of IOHK, I am also for it. I can imagine e.g PirlGuard solution in combination with the rough roadmap by IOHK, too.

**Comment by crazypool2019** [comment]
they should implement pirlguard now, asap, then they can see if they need anything else

**Comment by TheEnthusiasticAs** [comment]
Please don't mention the PirlGuard as a solution or any other not relevant one (see the topic headline). It was mentioned just as an example. Thank you.

**Comment by Dexaran** [comment]
I assume that the proposed `Securing Proof-of-Work Ledgers via Checkpointing` point of the roadmap is intended to solve the problem of 51% attacks. In this case discussing the alternative solutions should be legit, isn't it?

I'm not advocating anything described in here - just wondering how should we discuss the alternative solutions in order to understand if we are interested in ones described at this proposal or these are no longer releavant in presence of a better option.

**Comment by crazypool2019** [comment]
with a checkpoint you can rewrite the blockchain from that particular point, this solution does not prevent the attack, it just adds an additional difficulty

**Comment by gitr0n1n** [comment]
Adding my opinions:
TL;DR: I'm supportive of a treasury system for development.

Correct me if I'm wrong, but I believe Charles' 1 week timeframe was related to garnering community support for an embedded treasury system. The other topics appear to be his ideas related to solving the current state of the network. It appeared that his primary concern was that there would be funding for his developers if he committed resources to the ETC network. Again, correct me if I'm wrong please.

Treasury System:
I am a fan of a treasury system to support decentralized development on ETC. Having a treasury has the potential to attract dev talent that we don't even know about. We will likely see many more ECIPs with innovative tech for this specific application of blockchain; be it L1 or L2 proposals. I personally see a lot more pros than cons.

I am a fan of a funding system that's sole purpose is to advance the network itself. The Developers funding pool would be replenished by their ability to produce products that fill blocks (increased user usage). This appreciates the value prop of the network and aligns Developers and Miners incentives.

A L1 example, there is a community desire for Account Versioning and no present development talent to champion this idea. A treasury would incentivize developers to meet the unanswered needs of the non-developer community. Currently, the active development teams have no signaled intent to work on this highly requested feature that will add backwards compatibility to the network.

I agree with @Dexaran that L2 Smart Contracts can fund a majority of these funding mechs. But since ETC is still at that L1 focus, I'm not opposed to a baseline L1 portion of the inflation going to the developer. This will ensure that if blocks are being mined, development will be funded in a way that is solely reliant on the network (until the emission is 0); a big move away from third-party dependency (ETC Coop & ETC Labs models of funding development) which I view as a step in the right direction for decentralization of powers and segregation of duties in the ecosystem. Current material development funding comes from James Wo and Barry Silbert's private businesses. Let's add "the public ETC Network" to that list with a built in treasury system and remove third-party dependency for ETC development.

A L2 example, there is an entire DeFi ecosystem that can be ported over to ETC with a dev funding mech built into its foundation. Currently ETH DeFi teams are extorting the third-party ETC money to "Add ETC Support". Heresay: I believe Uniswap wanted something like a million dollars to add ETC, this is not a feasibly approach. Also it is not logical to pay extortion fees when this network can fund this development and link into existing networks due to the permissionless nature of blockchains.


Unknowns that I can not commit support for due to lack of knowledge and understanding at this moment:
- I do not know the mechanics of the governance system Charles discussed "Voltaire".
- I am hesitant to introduce a hybrid PoW/PoS model because I don't understand how this works yet. Also I am reluctant to any PoS due to so many poor implementations of it.

**Comment by BaikalMine** [comment]
How long will it take to develop this system? Until this system is developed - will we still suffer from 51% attacks or will we sit and mine the void without the ability to sell on exchanges?

**Comment by BaikalMine** [comment]
By introducing a checkpoint system, you force the pool owners to add a check for rollbacks, so that in the event of a rollback of the blockchain and a rollback of the pool balance, you do not pay the miners their balances that were accrued at the time of the block reorganization from their personal funds!

**Comment by mahofmahof** [comment]
ecip-1043 will boost hashrate for short term,dag is also not sustainable, ,on each increase lowers the hashrate for gpus too.

**Comment by mn851** [comment]
> Correct me if I'm wrong, but I believe Charles' 1 week timeframe was related to garnering community support for an embedded treasury system.

This is exactly what I took from the video. Ideas he presented were possible solutions on how the situation could be improved. I did not take them to be what he was suggesting as part of the 1 week to decide. Maybe this needs to be two separate ECIPs:

1) Short-term solutions to resolve network issues
2) Community support for an embedded treasury system with a 1-week decision timeline

**Comment by TheEnthusiasticAs** [comment]
There is a planned a Dev Call regarding the Ethereum Classic Treasury in #332

**Comment by q9f** [comment]
Closing this in favor of #332 for treasury and #333 for anything else going forward.
